modules/ai/assets/js/gutenberg/excerpt.js

```diff
@@ -0,0 +1,119 @@
+import React, { useRef } from 'react';
+import App from '../editor/app';
+import PropTypes from 'prop-types';
+import useUserInfo from '../editor/hooks/use-user-info';
+import useExcerptPrompt from '../editor/hooks/use-excerpt-prompt';
+import { useCallback, useEffect } from '@wordpress/element';
+
+const { useSelect, useDispatch } = wp.data;
+
+// Function to extract all textual content from a Gutenberg post
+const useGutenbergPostText = ( ) => {
+	const postContent = useSelect(
+		( select ) => select( 'core/editor' )?.getEditedPostContent(),
+		[],
+	);
+	const title = useSelect(
+		( select ) => select( 'core/editor' )?.getEditedPostAttribute( 'title' ),
+		[],
+	);
+
+	function extractTextFromGutenberg( content, extractedText ) {
+		// Create a temporary DOM element to parse the content
+		const tempDiv = document.createElement( 'div' );
+		tempDiv.innerHTML = content;
```

Comment:
Maybe:

```suggestion
		tempDiv.innerHTML = content;
		return tempDiv.innerText;
```

Reply:
it merges texts inside tables, quotas and other blocks into one long text

Reply:
It's OK. A post may contain mainly tables and quotes, and they should be part of the context.

Reply:
I meant it extracts it as one long text. Instead of multiple words Im getting one merged word, for some of the blocks like table blocks